sympy/functions/elementary/exponential.py

```diff
@@ -765,6 +766,11 @@ def _eval_expand_log(self, deep=True, **hints):
             p = perfect_power(int(arg))
             if p is not False:
                 return p[1]*self.func(p[0])
+            # expand as product of its prime factors if factor=True
+            if factor:
+                p = factorint(int(arg))
+                if int(arg) not in p.keys():
+                    return sum(log(arg / val**n) for val, n in p.items())
```

Comment:
Is this formula right?

Reply:
sorry, It should be 
`sum(n*log(val) for val, n in p.items())`

this should also remove the need for previous condition
`p = perfect_power(int(arg))`
`if p is not False: return p[1]*self.func(p[0])`